### ISSUE TITLE
HMR operation order issue

### ISSUE THREAD

**Issue by elsassph** [parcel-bundler/parcel#2488]
# 🐛 bug report

HMR may execute a changed module multiple times.

Example:
- parent A loads child B,
- both A and B change at once, leading to a single HMR update.

## 🤔 Expected Behavior

Expecting modules A and B to be evaluated only once.

## 😯 Current Behavior

1. module A is executed,
2. module B is executed,
3. module A is re-executed because is a parent of B.

## 💁 Possible Solution

Investigating `hmr-runtime.js` I found out that `hmrAccept` will always re-evalute a modules:
https://github.com/parcel-bundler/parcel/blob/master/packages/core/parcel-bundler/src/builtins/hmr-runtime.js#L162-L171

I suspect that HMR was designed for a single module change so it climbs the module hierarchy to re-execute the parents.

## 🔦 Context

Working on a new language support able to emit spilt bundles.
https://github.com/elsassph/parceljs-haxe-example/tree/wip

## 💻 Code Sample

https://github.com/elsassph/parceljs-haxe-example/blob/wip/script/HaxeAsset.js

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 1.x latest
| Node             | 10.x
| Operating System | Mac

**Comment by mischnic** [comment]
Duplicate of https://github.com/parcel-bundler/parcel/issues/2497
